doc/development/plugins.rst

```diff
@@ -358,6 +359,56 @@ object.
 As a convention, devices should do the translation and unpacking as late as possible in the function tree, and
 where possible pass the original :class:`~.wires.Wires` objects around.
 
+Device tracker support
+----------------------
+
+The device tracker stores and records information when tracking mode is turned on. Devices can store data like
+the number of executions, number of shots, number of batches, or remote simulator cost for users to interact with
+in a customizable way.
+
+Three aspects of the :class:`~.Tracker` class are relevant to plugin designers:
+
+* The boolean ``active`` attribute that denotes whether or not to update and record
+* ``update`` method which accepts keyword-value pairs and stores the information
+* ``record`` method which users can customize to log, print, or otherwise do something with the stored information
+
+To gain any of the device tracker functionality, a device should initialize with a placeholder 
+:class:`~.Tracker` instance. Users can overwrite this attribute by initializing a new instance with
+the device as an argument.
+
+We recommend placing the following code near the end of the ``execute`` method:
+
+.. code-block:: python
+
+  if self.tracker.active:
+    self.tracker.update(executions=1, shots=self._shots)
+    self.tracker.record()
+
+And similar code in the ``batch_execute`` method:
+
+.. code-block:: python
+
+  if self.tracker.active:
+    self.tracker.update(batches=1, batch_len=len(batches))
```

Comment:
Was `batches` defined in `batch_len=len(batches)`?

Reply:
oops... that's supposed to be `circuits`